feat(next): routes resolved hook (#10077)
withastro/docs: src/content/docs/en/guides/upgrade-to/v5.mdx

<ReadMore>Learn more about [the Adapter API](/en/reference/adapter-reference/) for building adapter integrations.</ReadMore>

### Deprecated: `routes` on `astro:build:done` hook (Integration API)

<SourcePR number="12329" title="feat(next): astro:routes:resolved"/>

In Astro v4.x, you could only get routes in an integration from the `astro:build:done` hook.

> Astro v5.0 deprecates the `routes` array passed to this hook. Instead, it exposes a new `astro:routes:resolved` hook that runs before `astro:config:done`, and whenever a route changes in development. It has all the same properties of the deprecated `routes` list, except `distURL` which is only available during build. For this, you can use the newly exposed `assets` map:

Reviewer: ```suggestion
Astro v5.0 deprecates the `routes` array passed to this hook. Instead, it exposes a new `astro:routes:resolved` hook that runs before `astro:config:done`, and whenever a route changes in development. It has all the same properties of the deprecated `routes` list, except `distURL` which is only available during build. 

#### What should I do?

Remove any instance of `routes` passed to `astro:build:done` and replace it with the new `astro:routes:resolved` hook. Access `distURL` on the newly exposed `assets` map:
```
Just needed something to fit into this pattern. Can you update the sentence above to make it correct/helpful?

<ReadMore>Learn more about [the Adapter API](/en/reference/adapter-reference/) for building adapter integrations.</ReadMore>

### Deprecated: `routes` on `astro:build:done` hook (Integration API)

<SourcePR number="12329" title="feat(next): astro:routes:resolved"/>

> In Astro v4.x, you could only get routes in an integration from the `astro:build:done` hook.

Reviewer: ```suggestion
In Astro v4.x, integrations accessed routes from the `astro:build:done` hook.
```
